pragma-node/src/handlers/entries/subscribe_to_entry.rs

```diff
@@ -0,0 +1,260 @@
+                let existing_pairs =
+                    only_existing_pairs(&state.timescale_pool, subscription_msg.pairs).await;
+                for pair_id in existing_pairs {
+                    if !subscribed_pairs.contains(&pair_id) {
+                        subscribed_pairs.push(pair_id.to_string());
+                    }
+                }
+            }
+            SubscriptionType::Unsubscribe => {
+                subscribed_pairs.retain(|pair| !subscription_msg.pairs.contains(pair));
+            }
+        };
+        if let Ok(ack_message) = serde_json::to_string(&SubscriptionAck {
+            msg_type: subscription_msg.msg_type,
+            pairs: subscribed_pairs.clone(),
+        }) {
+            if socket.send(Message::Text(ack_message)).await.is_err() {
+                let error_msg = "Message received but could not send ack message.";
+                send_error_message(socket, error_msg).await;
+            }
+        } else {
+            let error_msg = "Could not serialize ack message.";
+            send_error_message(socket, error_msg).await;
+        }
+    } else {
+        let error_msg = "Invalid message type. Please check the documentation for more info.";
+        send_error_message(socket, error_msg).await;
+    }
+}
+
+/// Send the current median entries to the client.
+async fn send_median_entries(
+    socket: &mut WebSocket,
+    state: &AppState,
+    subscribed_pairs: &[String],
+) -> Result<(), EntryError> {
+    if subscribed_pairs.is_empty() {
+        return Ok(());
+    }
+    let entries = match get_subscribed_pairs_entries(state, subscribed_pairs).await {
+        Ok(response) => response,
+        Err(e) => {
+            send_error_message(socket, &e.to_string()).await;
+            return Err(e);
+        }
+    };
+    if let Ok(json_response) = serde_json::to_string(&entries) {
+        if socket.send(Message::Text(json_response)).await.is_err() {
+            send_error_message(socket, "Could not send prices.").await;
+        }
+    } else {
+        send_error_message(socket, "Could not serialize prices.").await;
+    }
+    Ok(())
+}
+
+/// Get the current median entries for the subscribed pairs and sign them as Pragma.
+async fn get_subscribed_pairs_entries(
+    state: &AppState,
+    subscribed_pairs: &[String],
+) -> Result<SubscribeToEntryResponse, EntryError> {
+    let median_entries =
+        get_current_median_entries_with_components(&state.timescale_pool, subscribed_pairs)
+            .await
+            .map_err(|e| e.to_entry_error(&subscribed_pairs.join(",")))?;
+
+    let mut response: SubscribeToEntryResponse = Default::default();
+    let now = chrono::Utc::now().timestamp();
+    for entry in median_entries {
+        let median_price = entry.median_price.clone();
+        let mut oracle_price: AssetOraclePrice = entry
+            .try_into()
+            .map_err(|_| EntryError::InternalServerError)?;
+
+        let signature = sign_median_price_as_pragma(
```

Comment:
let's just use a generic `sign_data` function, as the params are given as args anyway. We can have a `sign_median_price_as_pragma` that would call the `sign_data` function